File: LinkedList/19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummyNode = new ListNode();
        dummyNode.next = head;
        ListNode tempNode = head;
        int size = 0;
        while(tempNode.next!=null){
            size++;
            tempNode = tempNode.next;
        }
        size++;
        tempNode = dummyNode;
        int counter = size - n;
        for(int i=0; i<counter; i++)
        {
            tempNode = tempNode.next;
        }
        if(tempNode.next == null){
            head=null;
            return head;
        }
        tempNode.next = tempNode.next.next;
        return dummyNode.next;
    }
}

File: Linked_List/19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        length = 0

        tmp = head

        while(tmp!=None):
            
            tmp = tmp.next
            length += 1


        counter = length - n

        dummy = ListNode()
        dummy.next = head

        tmp = dummy

        for i in range(counter):
            tmp = tmp.next

        tmp.next = tmp.next.next

        return dummy.next